core/src/main/scala/org/apache/spark/rdd/PairRDDFunctions.scala

```diff
@@ -1034,6 +1034,245 @@ class PairRDDFunctions[K, V](self: RDD[(K, V)])
     writer.commitJob()
   }
 
+  //////////////////////////////////////
+  // Async saveAsHadoopFile extension //
+  //////////////////////////////////////
+
+  /**
+   * Output the RDD to any Hadoop-supported file system, using a Hadoop `OutputFormat` class
+   * supporting the key and value types K and V in this RDD.
+   */
+  def saveAsHadoopFileAsync[F <: OutputFormat[K, V]](path: String)(
```

Comment:
Most of this section is just the change of the return type, with a FutureAction instead of Unit